src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js

```diff
@@ -110,66 +118,101 @@ const WorkflowDashboard = _.flow(
+              div({ style: styles.sectionTableLabel }, ['End:']), div([end ? Utils.makeCompleteDate(end) : 'N/A'])
+            ])
+          ]),
+          makeSection('Links', [
+            div({ style: { display: 'flex', flexFlow: 'row wrap', marginTop: '0.5rem' } }, [
+              h(Link, {
+                ...Utils.newTabLinkProps,
+                href: `${getConfig().jobManagerUrlRoot}/${workflowId}`,
+                style: { display: 'flex' },
+                tooltip: 'Job Manager'
+              }, [icon('tasks', { size: 18 }), ' Job Manager']),
+              h(Link, {
+                ...Utils.newTabLinkProps,
+                href: bucketBrowserUrl(`${bucketName}/${submissionId}/${workflowName}/${workflowId}`),
+                style: { display: 'flex', marginLeft: '1rem' },
+                tooltip: 'Execution directory'
+              }, [icon('folder-open', { size: 18 }), ' Execution Directory']),
+              h(Link, {
+                onClick: () => setShowLog(true),
+                style: { display: 'flex', marginLeft: '1rem' }
+              }, [icon('fileAlt', { size: 18 }), ' View execution log'])
+            ])
+          ])
+        ]),
+        failures && h(Collapse,
+          {
+            style: { marginBottom: '1rem' },
+            initialOpenState: true,
+            title: div({ style: Style.elements.sectionHeader }, [
+              'Workflow-Level Failures',
+              h(ClipboardButton, {
+                text: JSON.stringify(failures, null, 2),
+                style: { marginLeft: '0.5rem' },
+                onClick: e => e.stopPropagation() // this stops the collapse when copying
+              })
+            ])
+          }, [h(ReactJson, {
+            style: { whiteSpace: 'pre-wrap' },
+            name: false,
+            collapsed: 4,
+            enableClipboard: false,
+            displayDataTypes: false,
+            displayObjectSize: false,
+            src: restructureFailures(failures)
+          })]
+        ),
+        h(Collapse,
+          {
+            title: div({ style: Style.elements.sectionHeader }, ['Calls']),
+            initialOpenState: true
+          }, [
+            div({ style: { marginLeft: '1rem' } },
+              [makeSection('Total Call Status Counts', [
+                !_.isEmpty(calls) ? statusCell(workflow) : div({ style: { marginTop: '0.5rem' } }, ['No calls have been started by this workflow.'])
+              ]),
+              !_.isEmpty(calls) && makeSection('Call Lists', [
+                _.map(callName => {
+                  return h(Collapse, {
+                    key: callName,
+                    style: { marginLeft: '1rem', marginTop: '0.5rem' },
+                    title: div({ style: { ...Style.codeFont, ...Style.elements.sectionHeader } }, [`${callName} × ${calls[callName].length}`]),
+                    initialOpenState: !_.isEmpty(_.filter(c => { return c.executionStatus !== 'Done' })(calls[callName]))
```

Comment:
I think this could be simplified to:
```suggestion
                    initialOpenState: !_.every({ executionStatus: 'Done' }, calls[callName])
```
(with the bonus that it will perform better because `_.every` will halt as soon as it finds a false case)

Reply:
Much nicer! I also didn't realize that the destructuring syntax could be used as predicate functions. Very very nice!

Reply:
It's not destructuring, we're actually passing an object there. It's a lodash-specific feature: anywhere you pass a predicate function, you can substitute an object, and lodash will interpret it as "for each key in the passed object, test that the item has a matching value at that key"